x-pack/solutions/observability/plugins/apm/ui_tests/fixtures/page_objects/service_map.ts

```diff
@@ -5,36 +5,52 @@
  * 2.0.
  */
 
-import type { KibanaUrl, ScoutPage } from '@kbn/scout';
+import type { KibanaUrl, Locator, ScoutPage } from '@kbn/scout-oblt';
+import { expect } from '@kbn/scout-oblt';
 
 export class ServiceMapPage {
-  constructor(private readonly page: ScoutPage, private readonly kbnUrl: KibanaUrl) {}
+  public serviceMap: Locator;
+  public zoomInBtn: Locator;
+  public zoomOutBtn: Locator;
+  public centerServiceMapBtn: Locator;
+  public noServicesPlaceholder: Locator;
 
-  async waitForPageToLoad() {
-    await this.page.waitForSelector(
-      '[data-test-subj="kbnAppWrapper visibleChrome"] [aria-busy="false"]',
-      { state: 'visible' }
-    );
+  constructor(private readonly page: ScoutPage, private readonly kbnUrl: KibanaUrl) {
+    this.serviceMap = page.testSubj.locator('serviceMap');
+    this.zoomInBtn = page.locator('button[aria-label="Zoom in"]');
+    this.zoomOutBtn = page.locator('button[aria-label="Zoom out"]');
+    this.centerServiceMapBtn = page.testSubj.locator('centerServiceMap');
+    this.noServicesPlaceholder = page.locator('.euiEmptyPrompt__content .euiTitle');
   }
 
   async gotoWithDateSelected(start: string, end: string) {
-    this.page.goto(`${this.kbnUrl.app('apm')}/service-map?&rangeFrom=${start}&rangeTo=${end}`);
-    await this.waitForPageToLoad();
+    await this.page.goto(
+      `${this.kbnUrl.app('apm')}/service-map?&rangeFrom=${start}&rangeTo=${end}`
+    );
+    return this.page.waitForLoadingIndicatorHidden();
   }
+
   async gotoDetailedServiceMapWithDateSelected(start: string, end: string) {
-    this.page.goto(
+    await this.page.goto(
       `${this.kbnUrl.app(
         'apm'
       )}/services/opbeans-java/service-map?&rangeFrom=${start}&rangeTo=${end}`
     );
-    await this.waitForPageToLoad();
+    return this.page.waitForLoadingIndicatorHidden();
```

Comment:
we extended Playwright Page with `ScoutPage` class, that provides KIbana specific helpers like `goToApp` , `waitForLoadingIndicatorHidden`, etc